crates/iceberg/src/scan.rs

```diff
@@ -180,7 +185,42 @@ pub type ArrowRecordBatchStream = BoxStream<'static, crate::Result<RecordBatch>>
 impl FileScanTask {
     /// Returns a stream of arrow record batches.
     pub async fn execute(&self) -> crate::Result<ArrowRecordBatchStream> {
-        todo!()
+        match self.data_file.content_type() {
```

Comment:
After checking java/python's implementation, I think it's in appropriate to have this api. To convert a `FileScanTask` to arrow record batch stream, we may need some other configurations:
1. Record batch size, e.g. number of records of each record batch, this may affect memory usage of upstream executors
2. `FileIO`, as in your implementation.

Also it's more general to have multi `FileScanTask`s in size based reader.

Reply:
And seems it should not contain `FileIO` in FileScanTask if we want it can be de/serde. Maybe we can get a stream of FileScanTask and use FileIO and FileScanTask to construct the reader.

Reply:
FileScanTask no longer has ref to fileio in latest version